ebs/error_budget.py

```diff
@@ -417,6 +410,407 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
             self.output_to_json(output_filename_prefix+'.json')
 
 
+class ErrorBudgetMcmc(object):
+
+    def __init__(self, config_file="config.yml"):
+        with open(config_file, 'r') as config:
+            self.config = yaml.load(config, Loader=yaml.FullLoader)
+        self.target_list = None
+        self.exo_zodi = None
+        self.eeid = None
+        self.eepsr = None 
+        self.wfe = None
+        self.wfsc_factor = None
+        self.sensitivity = None
+        self.post_wfsc_wfe = None
+        self.angles = None
+        self.contrast = None
+        self.working_angles = []
+        self.QE = None
+        self.sread = None
+        self.idark = None
+        self.Rs = None
+        self.lensSamp = None
+        self.pixelNumber = None
+        self.pixelSize = None
+        self.optics = None
+        self.BW = None
+        self.IWA = None
+        self.throughput = None
+        self.core_mean_intensity = None
+        self.SNR = None
+        self.ppFact_filename = None
+        self.contrast_filename = None
+        self.throughput_filename = None
+        self.ppFact_filename = None
+        self.exosims_pars_dict = None
+        self.trash_can = []
+
+    @property
+    def delta_contrast(self):
+        """
+        Compute change in contrast due to residual WFE and assign the array to 
+        `self.ppFact`. 
+
+        Reference
+        ---------
+        - See <Post-Processing Factor> document for mathematical description
+
+        """
+        if (self.wfe is not None and self.wfsc_factor is not None 
```

Comment:
suggestion for a little more concise way to do this 

`if any([val is None for val in [self.wfe, self.wfsc_factor, self.sensitivity, self.contrast])`